tests/test-pow.c

```diff
@@ -49,31 +81,58 @@ int main()
 
     int mwm = 14;
 
-    /* test implementation of the PoW FPGA accelerator with mwm = 14 */
-    initializeImplContext(&PoWFPGAAccel_Context);
-    void *pow_ctx =
-        getPoWContext(&PoWFPGAAccel_Context, (int8_t *) trytes, mwm);
-    assert(pow_ctx);
-    doThePoW(&PoWFPGAAccel_Context, pow_ctx);
-    int8_t *ret_trytes = getPoWResult(&PoWFPGAAccel_Context, pow_ctx);
-    assert(ret_trytes);
-    freePoWContext(&PoWFPGAAccel_Context, pow_ctx);
-
-    Trytes_t *trytes_t = initTrytes(ret_trytes, 2673);
-    assert(trytes_t);
-    Trytes_t *hash_trytes = hashTrytes(trytes_t);
-    assert(hash_trytes);
-
-    /* Validation */
-    Trits_t *ret_trits = trits_from_trytes(hash_trytes);
-    for (int i = 243 - 1; i >= 243 - mwm; i--) {
-        assert(ret_trits->data[i] == 0);
-    }
+    ImplContext ImplContextArr[ ] = {
+        #if defined(ENABLE_AVX)
+        PoWAVX_Context,
+        #elif defined(ENABLE_SSE)
+        PoWSSE_Context,
+        #else
+        PoWC_Context,
+        #endif
+
+        #if defined(ENABLE_OPENCL)
+        PoWCL_Context,
+        #endif
+
+        #if defined(ENABLE_FPGA_ACCEL)
+        PoWFPGAAccel_Context,
+        #endif
+    };
 
-    free(ret_trytes);
-    freeTrobject(trytes_t);
-    freeTrobject(hash_trytes);
-    freeTrobject(ret_trits);
+
+    for (int idx = 0; idx < sizeof(ImplContextArr)/sizeof(ImplContext); idx++) {
+        printf("%s\n",description[idx]);
+
+        ImplContext *PoW_Context_ptr = &ImplContextArr[idx];
+
+        /* test implementation with mwm = 14 */
+        initializeImplContext(PoW_Context_ptr);
+        void *pow_ctx = getPoWContext(PoW_Context_ptr, (int8_t *) trytes, mwm);
+        assert(pow_ctx);
+        doThePoW(PoW_Context_ptr, pow_ctx);
+        int8_t *ret_trytes = getPoWResult(PoW_Context_ptr, pow_ctx);
+        assert(ret_trytes);
+        freePoWContext(PoW_Context_ptr, pow_ctx);
+
+        Trytes_t *trytes_t = initTrytes(ret_trytes, 2673);
```

Comment:
Replace the constant `2673` with marco defined in `constants.h`

Reply:
No macro is equal to `2673` in `constants.h`.
However, there exists a macro `#define TRANSACTION_LENGTH 2673 * 3`.

Should I change the macro like this
```
#define TRANSACTION_TRYTES_LENGTH 2673
...
#define TRANSACTION_TRITS_LENGTH TRANSACTION_TRYTES_LENGTH * 3
```
or is there any suggestion?

Reply:
I agree with the proposed changes by @marktwtn. Also, we can follow file `trit_tryte.h` in [entangled](https://github.com/iotaledger/entangled).